lib/iris/cube.py

```diff
@@ -751,10 +751,13 @@ def __init__(self, data, standard_name=None, long_name=None,
         # Cell Measures
         self._cell_measures_and_dims = []
 
+        # We need to set the dtype before the fill_value,
+        # as the fill_value is checked against self.dtype.
+        self._dtype = None
+        if dtype is not None:
+            self.dtype = dtype
```

Comment:
I don't think we need to have this check here, we just need to initialise `_dtype` and then set `self.dtype = dtype`